File: vacation/vacation/VacationApproved.go
package vacation

import (
	"time"
)

type VacationApproved struct{
	EventType string	`json:"eventType" type:"string"`
	TimeStamp string 	`json:"timeStamp" type:"string"`
	Id string `json:"id" type:"string"` 
	StartDate  `json:"startDate" type:""` 
	EndDate  `json:"endDate" type:""` 
	Reason string `json:"reason" type:"string"` 
	
}

func NewVacationApproved() *VacationApproved{
	event := &VacationApproved{EventType:"VacationApproved", TimeStamp:time.Now().String()}

	return event
}

File: vacation/src/main/java/human/resource/mgmt/domain/VacationApproved.java
package human.resource.mgmt.domain;

import human.resource.mgmt.domain.*;
import human.resource.mgmt.infra.AbstractEvent;
import java.util.*;
import lombok.*;

@Data
@ToString
public class VacationApproved extends AbstractEvent {

    private String id;
    private Date startDate;
    private Date endDate;
    private String reason;

    public VacationApproved(Vacation aggregate) {
        super(aggregate);
    }

    public VacationApproved() {
        super();
    }
}
